packages/grid/src/vaadin-grid-selection-mixin.js

```diff
@@ -22,6 +22,24 @@ export const SelectionMixin = (superClass) =>
           sync: true,
         },
 
+        /**
+         * A function to check whether a specific item in the grid may be
+         * selected or deselected by the user. Used by the selection column to
+         * conditionally enable to disable checkboxes for individual items. This
+         * function does not prevent programmatic selection/deselection of
+         * items. Changing the function does not modify the currently selected
+         * items.
+         *
+         * Receives an item instance and should return a boolean indicating
+         * whether users may change the selection state of that item.
+         *
+         * @type {(item: !GridItem) => boolean}
+         */
+        isItemSelectable: {
```

Comment:
Other similar APIs, such as part name generator or `isCellEditable`, use a row model instance (see [`__getRowModel`](https://github.com/vaadin/web-components/blob/79ecca94f4b0dca53e860100b2b0ccaea1b42e42/packages/grid/src/vaadin-grid-mixin.js#L1024C1-L1039C6)). However that only works if you have a row instance available, whereas this can also be called for items that don't have a rendered row. Not sure if we need to align this.

Reply:
Discussed this and decided to only pass the item. Technically we could construct a row model object from the item itself before passing it to the function, but that would potentially require iterating over all grid caches to figure out  the flat row index, or whether an item is a nested one. If we later on want to add support for select all, we'll have to run this function in some way or another over all grid items, at which point this could become a performance issue.